com.unity.ml-agents/Runtime/Analytics/TrainingAnalytics.cs

```diff
@@ -0,0 +1,155 @@
+using System;
+using System.Collections.Generic;
+using Unity.Barracuda;
+using Unity.MLAgents.Actuators;
+using Unity.MLAgents.Inference;
+using Unity.MLAgents.Policies;
+using Unity.MLAgents.Sensors;
+using UnityEngine;
+using UnityEngine.Analytics;
+
+#if UNITY_EDITOR
+using UnityEditor;
+using UnityEditor.Analytics;
+#endif
+
+namespace Unity.MLAgents.Analytics
+{
+    internal class TrainingAnalytics
+    {
+        const string k_VendorKey = "unity.ml-agents";
+        const string k_TrainingEnvironmentInitializedEventName = "ml_agents_trainingenvironmentinitialized";
+        const string k_BehaviorTrainingStartedEventName = "ml_agents_behaviortrainingstarted";
+        const string k_RemotePolicyStartedEventName = "ml_agents_remotepolicystarted";
+
+        private static readonly string[] s_EventNames =
+        {
+            k_TrainingEnvironmentInitializedEventName,
+            k_BehaviorTrainingStartedEventName,
+            k_RemotePolicyStartedEventName
+        };
+
+        /// <summary>
+        /// Whether or not we've registered this particular event yet
+        /// </summary>
+        static bool s_EventsRegistered = false;
+
+        /// <summary>
+        /// Hourly limit for this event name
+        /// </summary>
+        const int k_MaxEventsPerHour = 1000;
+
+        /// <summary>
+        /// Maximum number of items in this event.
+        /// </summary>
+        const int k_MaxNumberOfElements = 1000;
+
+        /// <summary>
+        /// Behaviors that we've already sent events for.
+        /// </summary>
+        private static HashSet<string> s_SentRemotePolicyStartedBehaviors;
+
+        private static Guid s_TrainingSessionId;
+
+        static bool EnableAnalytics()
+        {
+            if (s_EventsRegistered)
+            {
+                return true;
+            }
+
+            foreach (var eventName in s_EventNames)
+            {
+#if UNITY_EDITOR
+                AnalyticsResult result = EditorAnalytics.RegisterEventWithLimit(eventName, k_MaxEventsPerHour, k_MaxNumberOfElements, k_VendorKey);
+#else
+                AnalyticsResult result = AnalyticsResult.UnsupportedPlatform;
+#endif
+                if (result != AnalyticsResult.Ok)
+                {
+                    return false;
+                }
+
+            }
+            s_EventsRegistered = true;
+
+            if (s_SentRemotePolicyStartedBehaviors == null)
+            {
+                s_SentRemotePolicyStartedBehaviors = new HashSet<string>();
+                s_TrainingSessionId = Guid.NewGuid();
+            }
+
+            return s_EventsRegistered;
+        }
+
+        public static bool IsAnalyticsEnabled()
+        {
+#if UNITY_EDITOR
+            return EditorAnalytics.enabled;
+#else
+            return false;
+#endif
+        }
+
+        public static void RemotePolicyStarted(
```

Comment:
I do not see where the other events (Behavior and Environment Initialization) are. Are they not here yet or did I miss them? I ask because it seems to me like there could be some code we could reuse.

Reply:
Not here yet. The EnableAnalytics() call will initialize all of them.

Reply:
Added placeholder events for these. Still waiting to finalize the fields at a high level before setting them here.